src/blockdata/script.rs

```diff
@@ -268,13 +271,21 @@ impl Script {
     }
 
     /// Generates P2WPKH-type of scriptPubkey
-    pub fn new_v0_wpkh(pubkey_hash: &WPubkeyHash) -> Script {
-        Script::new_witness_program(WitnessVersion::V0, &pubkey_hash.to_vec())
+    pub fn new_wpkh(pubkey_hash: &WPubkeyHash) -> Script {
```

Comment:
Is there any special reason to make this breaking change? I really liked the `v0` part, but I won't die on that hill :)

Reply:
The PR is breaking anyway at least because it is based on WitnessVersion PR which is highly intrusive. And even if we will have `v5_` one day, it still will be a breaking change (caused by bitcoin softfork). 

As of today, keeping `v0_` seems to be pointless. If one day with `v5_wpkh` we will need it once more, again, that will be a breaking change and we will introduce `v0_` once again.

Reply:
Why would a soft fork be a breaking change?

Reply:
not soft fork, but existing API changes merged into rust-bitcoin after lat release

Reply:
Do we want to cause breaking changes every time there's a soft fork though? I'd like this crate to be stabilized at some point...

Reply:
It is absolutely unrelated. API breaks nearly every release for now and current master is already requires major update.

Reply:
I don't think current situation should stay as is but I guess a new issue is better for this discussion: https://github.com/rust-bitcoin/rust-bitcoin/issues/649

Reply:
I am also confused about this. The body of the function clearly hardcodes V0. Why would the name of the function not reflect that.?

Reply:
@apoelstra my take is that there is no other P2WPKH than this one (with v0) and we do not put `v1` in `to_tr` case. So either we should have `to_v0_wpkh`/`to_v1_tr` or `to_wpkh`/`to_tr` and the later seems more readable.

Reply:
Hmm, `to_v1_tr` doesn't sound too bad to me.

Reply:
Ok, returned `v0` back :)

Reply:
Actually we have API inconsistency, when methods in `Address` do not have `v0`/`v1` prefix, while methods in `Script` do have. I think we need to stick to a single way of naming @apoelstra @Kixunil

Reply:
I personally like the explicitness of including version number but don't consider it a huge issue. If someone feels strongly about it go ahead and ignore me.

Reply:
We can add `v1_wpkh` as an alias for `tr` if it makes people happy. The reason to have v0 in one case is that "wpkh" does not imply a version, while `tr` does.

Reply:
To be clear, I'm fine with the current solution (keeping v0 and also having `v1_tr`). I just don't see the old situation as inconsistent.